### PR TITLE
LPS-167142 [Raylife AP] Fill information with Objects data

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [LPS-167142](https://issues.liferay.com/browse/LPS-167142)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @liferay/empty-line-between-elements */
+/* eslint-disable @liferay/no-abbreviations */
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:2]: Is this really necessary?

**Review by joanagrego — APPROVED** [review]
Approved! Thank you @eliasantosf :white_check_mark:

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 1 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:2]: Yes. I need to concatenate a space between two strings.

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:relevant

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 2]
```diff
@@ -14,19 +15,52 @@
...
 import './index.scss';
 
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 
 import NavigationBar from '../../../../../common/components/navigation-bar';
 import {getWebDavUrl} from '../../../../../common/utils/webdav';
 
-const PolicyDetail = () => {
+type PolicyDetails = {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:23]: Please, change it.
```suggestion
type PolicyDetailsType = {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 3]
```diff
@@ -14,19 +15,52 @@
...
+	gender: string;
+	highestEducation: string;
+	maritalStatus: string;
+	model: string;
+	occupation: string;
+	ownership: string;
+	primaryUsage: string;
+};
+
+const PolicyDetail = ({application}: any) => {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:37]: Please, advise the correct type for it.
```suggestion
const PolicyDetail = ({application}: ???) => {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 4]
```diff
@@ -14,19 +15,52 @@
...
 	enum navBarLabels {
 		Vehicles = 'Vehicles',
 		Drives = 'Drives',
 	}
-	const amountOfPolicyDetail = [0];
 	const navbarLabel = [navBarLabels.Vehicles, navBarLabels.Drives];
 	const [active, setActive] = useState(navbarLabel[0]);
+	const [applicationData, setApplicationData] = useState<any>();
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:44]: Please, advise the correct type for it.
```suggestion
	const [applicationData, setApplicationData] = useState<???>();
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 5]
```diff
@@ -42,98 +76,149 @@ const PolicyDetail = () => {
...
-									</div>
+			{active === navBarLabels.Vehicles &&
+				applicationData?.vehicleInfo?.form.map(
+					(curentVehicle: PolicyDetails, indexVehicle: number) => (
+						<div
+							className="bg-neutral-0 pl-6 policy-detail-border pr-6 pt-6"
+							key={indexVehicle}
+						>
+							<div className="d-flex flex-row flex-wrap justify-content-between">
+								{indexVehicle !== 0 && (
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:87]: Please, change it.
```suggestion
								{indexVehicle && (
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 6]
```diff
@@ -42,98 +76,149 @@ const PolicyDetail = () => {
...
+			{active === navBarLabels.Drives &&
+				applicationData?.driverInfo?.form.map(
+					(curentDriver: PolicyDetails, indexDriver: number) => (
+						<div
+							className="bg-neutral-0 pl-6 policy-detail-border pr-6 pt-6"
+							key={indexDriver}
+						>
 							<div className="d-flex flex-row flex-wrap justify-content-between">
-								{indexPolicyDetail !== 0 && (
+								{indexDriver !== 0 && (
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:166]: Please, change it.
```suggestion
								{indexDriver && (
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 5 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:87]: indexVehicle and indexDriver is not boolean value, it's a number. If I check this way the numeric value is rerendered as one of the columns of the content. Any other suggestions for this case?

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 3 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:37]: This object is already prefilled as any in the prop, should I define a type for it anyway?

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 4 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:55]: All ApplicationData attributes are typed in PolicyDetailsType and ApplicationPolicyDetailsType.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 3 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 2 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:forward

**Pull request closed without merging**